pkg/controller/cluster/cluster.go

```diff
@@ -66,15 +66,22 @@ type ClusterReconciler struct {
 	Client          client.Client
 	Scheme          *runtime.Scheme
 	record.EventRecorder
-	SharedAgentImage           string
-	SharedAgentImagePullPolicy string
-	K3SImage                   string
-	K3SImagePullPolicy         string
-	PortAllocator              *agent.PortAllocator
+	SharedAgentImage            string
+	SharedAgentImagePullPolicy  string
+	SharedAgentImageRegistry    string
+	VirtualAgentImage           string
+	VirtualAgentImagePullPolicy string
+	VirtualAgentImageRegistry   string
+	K3SServerImage              string
+	K3SServerImagePullPolicy    string
+	K3SServerImageRegistry      string
+	PortAllocator               *agent.PortAllocator
+	ServerImagePullSecrets      []string
+	AgentImagePullSecrets       []string
 }
 
 // Add adds a new controller to the manager
-func Add(ctx context.Context, mgr manager.Manager, sharedAgentImage, sharedAgentImagePullPolicy, k3SImage string, k3SImagePullPolicy string, maxConcurrentReconciles int, portAllocator *agent.PortAllocator, eventRecorder record.EventRecorder) error {
+func Add(ctx context.Context, mgr manager.Manager, sharedAgentImage, sharedAgentImagePullPolicy, sharedAgentImageRegistry, virtualAgentImage, virtualAgentImagePullPolicy, virtualAgentImageRegistry, k3SServerImage, k3SServerImagePullPolicy, k3SServerImageRegistry string, maxConcurrentReconciles int, portAllocator *agent.PortAllocator, eventRecorder record.EventRecorder, serverImagePullSecrets, agentImagePullSecrets []string) error {
```

Comment:
This func is getting way too long (16 arguments). It's time to refactor it. :smile: